cms/src/main/java/uk/nhs/digital/ps/modules/searchableFlagModule/DatasetProcessor.java

```
@@ -0,0 +1,74 @@
+            // no parent publication found
+            log.error("Parent publication not found");
+            log.error("failed for node" + getRepoValue(node::getPath));
+            return;
+        }
+
+        // get common:searchable value
+        boolean searchable = getRepoValue(() -> parentPublication.getNode(parentPublication.getName()).getProperty("publicationsystem:PubliclyAccessible").getBoolean());
+
+        // apply it to dataset document(s)
+        streamChildNodes(node)
+            .filter((document) -> hasValidStateAndType(document, validStatuses,"publicationsystem:dataset"))
+            .forEach((document) -> setSearchableFlag(document, searchable));
+    }
+
+    private Node findParentPublication(Node node) {
+        Node publication = null;
+
+        try {
+            Node folder = node.getParent();
+            publication = findPublicationInFolder(folder);
+        } catch (RepositoryException ex) {
+            ex.printStackTrace();
+        }
+
+        return publication;
+    }
+
+    private Node findPublicationInFolder(Node node) throws RepositoryException {
+        for (NodeIterator i = node.getNodes(); i.hasNext(); ) {
+            Node object = i.nextNode();
+            if (isParentPublication(object)) {
+                return object;
+            }
+        }
+
+        if (node.getSession().getRootNode().equals(node)) {
+            return null;
+        }
+
+        return findPublicationInFolder(node.getParent());
+    }
+
+    private boolean isParentPublication(Node node) throws RepositoryException {
+        return "content".equals(node.getName()) && "publicationsystem:publication".equals(node.getNode(node.getName()).getPrimaryNodeType().getName());
+    }
+
+    private void setSearchableFlag(Node document, boolean searchable) {
+        try {
+            document.setProperty(SEARCHABLE_FLAG, searchable);
+        } catch (RepositoryException ex) {
+            ex.printStackTrace();
```

Either throw or log rather than printStackTrace